proc_macros/src/rule.rs

```diff
@@ -127,6 +127,15 @@ struct RuleListenerSpec {
     callback: ExprClosure,
 }
 
+impl RuleListenerSpec {
```

Comment:
Macro "magic" for deciding whether this listener is "single-capture-oriented" vs "match-oriented" based on whether it names its first param `node` vs `captures`